tests/vmi_lifecycle_test.go

```diff
@@ -296,8 +296,6 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 				By("Checking console text")
 				err = tests.CheckForTextExpecter(vmi, []expect.Batcher{
-					&expect.BSnd{S: "\n"},
-					&expect.BSnd{S: "\n"},
 					&expect.BExp{R: consoleText},
```

Comment:
not sure we want to eliminate both, at least one \r as most of the place do, 
well as long as each batch finishes in a prompt , it wont be needed, but since most still have \n (or \r) in the begining
i would stick to it

Reply:
Can you please explain what sending those `\n`s gives?
`&expect.BExp{R: consoleText}` is looking for output from the startup.

The new `ExpectBatchWuthSendValidation` will be flaky if we will keep those `\n` sends. Since it assumes the `expect.BExp` should be the result (and as so, appear after) the `expect.BSnd`. Since `consoleText` is not the result of `\n`, it has to be eliminated.

Reply:
> Can you please explain what sending those `\n`s gives?

It make sure that this batch isnt depend on the previous one gracefulStop, up to you, we dont have it on some other places still, unless you changed them , if you didnt ,it means that if someone try to adapt them to ExpectBatchWithSendValidation, he will have problems, 

that being sad i would suggest if possible please to create a how to list with the known culprits and guides?

> `&expect.BExp{R: consoleText}` is looking for output from the startup.
> 
> The new `ExpectBatchWuthSendValidation` will be flaky if we will keep those `\n` sends. Since it assumes the `expect.BExp` should be the result (and as so, appear after) the `expect.BSnd`. Since `consoleText` is not the result of `\n`, it has to be eliminated.

Understood, 
just a nit i would suggest hat ExpectBatchWithSendValidation will enforce to start with Snd but not mandatory or maybe not good by design, if you want to keep the freedom still.

Reply:
i think i saw once that you do need to send \n
which indeed the result is prompt
(like doing enter)
else you might have a race that the buffer is empty and you will wait forever without it ?
even that the previous expecter was done gracefully

this is also what happened at least on most of the places
that it start by sending \n and expecting prompt

u even use still CheckForTextExpecter here dont you ?

Reply:
My mistake, the \n may be needed, depends on timing (the required `consoleText` is printed during the startup and also after the startup finished if `\n` is sent. But since the epxpecter may be created before/during or after the startup we cannot rely on text printed during the startup).
I also updated `ExpectBatchWithSendValidation` to return an error if the batch doesn't start with a send.